lib/src/from_handlers.dart

```diff
@@ -67,12 +67,9 @@ class _StreamTransformer<S, T> implements StreamTransformer<S, T> {
       controller.onResume = () => subscription?.resume();
     }
     controller.onCancel = () {
-      if (controller.hasListener || subscription == null) {
-        return new Future.value();
-      }
-      var toCancel = subscription;
+      if (controller.hasListener || subscription == null) return;
+      subscription.cancel();
```

Comment:
I think I found an approach that won't lose this Future. I shouldn't need to cancel the subscription if the `onCancel` is triggered by the stream finishing - so in the case where the values are done I'll skip it, otherwise I'll use that as the return value.